In [13]:
import re
import math
import pickle
import random
import pandas as pd

from indic_transliteration import sanscript
from indic_transliteration.sanscript import SchemeMap, SCHEMES, transliterate

In [15]:
#Data Lists
nums = [0,1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19,20,30,40,50,60,70,80,90,100]

hindiN = ['शून्य','एक','दो' ,'तीन' ,'चार' ,'पांच','छः','सात','आठ','नौ','दस','ग्यारह','बारह','तेरह','चौदह','पंद्रह','सोलह','सत्रह','अट्ठारह','उन्निस','बीस','इक्कीस','बाईस','तेईस','चौबीस','पच्चीस','छब्बीस','सत्ताईस','अट्ठाईस','उनतीस','तीस','इकतीस','बत्तीस','तैंतीस','चौंतीस','पैंतीस','छ्त्तीस','सैंतीस','अड़तीस','उनतालीस','चालीस','इकतालीस','बयालीस','तैंतालीस','चौंतालीस','पैंतालीस','छियालीस','सैंतालीस','अड़तालीस','उनचास','पचास','इक्याबन','बावन','तिरेपन','चौबन','पचपन','छप्पन','सत्तावन','अट्ठावन','उनसठ','साठ','इकसठ','बासठ','तिरसठ','चौंसठ','पैंसठ','छियासठ','सड़सठ','अड़सठ','उनहत्तर','सत्तर','इकहत्तर','बहत्तर','तिहत्तर','चौहत्तर','पचहत्तर','छिहत्तर','सतहत्तर','अठहत्तर','उनासी','अस्सी','इक्यासी','बयासी','तिरासी','चौरासी','पचासी','छियासी','सतासी','अठासी','नवासी','नब्बे','इक्यानबे','बानवे','तिरानवे,','चौरानवे','पचानवे' ,'छियानवे','सत्तानवे' ,'अट्ठानवे' ,'निन्यानवे' ,'सौ']

hindiN = ['शून्य','एक','दो' ,'तीन' ,'चार' ,'पांच','छः','सात','आठ','नौ','दस','ग्यारह','बारह','तेरह','चौदह','पंद्रह','सोलह','सत्रह','अट्ठारह','उन्निस','बीस','तीस','चालीस','पचास','साठ','सत्तर','अस्सी','नब्बे','सौ']

marathiN = ['शून्य', 'एक', 'दोन', 'तीन', 'चार', 'पाच', 'छः', 'सात', 'आठ', 'नौ', 'दहा', 'ग्यारह', ' बारह ',' तेरह ',' चौधह ',' पंद्रह ',' सोलह ',' सत्रह ',' अट्ठरह ',' उन्निस ',' बीस ',' तिस ',' चालीस ',' पचास ',' साठ ' , 'सत्तार', 'अस्सी', 'नब्बे', 'सौ']

teluguN = ['సున్న','ఒకటి','రెండు','మూడు','నాలుగు','అయిదు','ఆరు','ఏడు','ఎనిమిది','తొమ్మిది','పది','పదకొండు','పన్నెండు','పదమూడు','పధ్నాలుగు','పదునయిదు','పదహారు','పదిహేడు','పధ్ధెనిమిది','పందొమ్మిది','ఇరవై','ముప్పై','నలభై','యాభై','అరవై','డెబ్బై','ఎనభై','తొంభై','వంద']

letters = ['a','b','c','d','e','f','g','h','i','j','k','l','m','n','o','p','q','r','s','t','u','v','w','x','y','z']
teluguL = ['ఎ','బి', 'సి', 'డి', 'ఈ', 'ఎఫ్', 'జి', 'హెచ్', 'అయి', 'జె', 'కె', 'ఎల్', 'ఎం', 'ఎన్', 'ఓ', 'పి', 'క్యు', 'అర్', 'ఎస్', 'టి', 'యు', 'వి', 'డబల్యు', 'ఎక్స్', 'వయి', 'జెడ్']
hindiL = ['ए', 'बी', 'सी', 'डी', 'ई', 'एफ', 'जी', 'एच', 'आई', 'जे', 'के', 'एल', 'एम','एन', 'ओ', 'पी', 'क्यू', 'आर', 'एस', 'टी', 'यू', 'वी', 'डब्ल्यू', 'एक्स', 'वाई', 'जेड']
marathiL = ['ए', 'बी', 'सी', 'डी', 'ई', 'एफ', 'जी', 'एच', 'आई', 'जे', 'के', 'एल', 'एम','एन', 'ओ', 'पी', 'क्यू', 'आर', 'एस', 'टी', 'यू', 'वी', 'डब्ल्यू', 'एक्स', 'वाई', 'जेड']



#Dictionaries
telugu_nums = {nums[i]:teluguN[i] for i in range(len(nums))}
hindi_nums = {nums[i]:hindiN[i] for i in range(len(nums))}
marathi_nums = {nums[i]:marathiN[i] for i in range(len(nums))}

telugu_letters ={letters[i]:teluguL[i] for i in range(len(letters))} 
hindi_letters ={letters[i]:hindiL[i] for i in range(len(letters))} 
marathi_letters ={letters[i]:marathiL[i] for i in range(len(letters))} 

#Special Lists
sp_nums = ['', 'ఒక','ఇద్దరు','ముగ్గురు','నలుగురు','ఐదుగురు','ఆరుగురు','ఏడుగురు']
sp_nums = ['', '','','','','','','']

telegu_class_nums = ['', 'ఒకటవ', 'రెండవ', 'మూడవ', 'నాలుగవ', 'ఐదవ', 'ఆరవ', 'ఏడవ', 'ఎనిమిదవ', 'తొమ్మిదవ', 'పదవ', 'పదకొండవ', 'పన్నెండవ']

hindi_class_nums = ['', 'प्रथम', ' ूसरी', 'तीसरी', 'चौथी', 'पाँचवी', 'छठी', 'सातवीं', 'आठवीं', 'नौवीं', 'दसवीं', 'ग्यारहवीं', 'बारहवीं']

marathi_class_nums = ['', 'पहिला','दुसरा','तीसरी','चौथी', 'पाचवा', 'छठी', 'सहावा', 'आठवीं', 'नौवीं', 'दहावीं', 'अकरावीं', 'बारहवीं']

class_nums = marathi_class_nums

# Known Phrases
teluguWord = {"anantapur": "అనంతపూర్", "guntur": "గుంటూర్", "chitoor": "చిత్తూరు", "rural": "రూరల్ ", "telugu": "తెలుగు", "english": "ఇంగ్లీష్", "kannada":"కన్నడ"}

hindiWord = {"anantapur": "अनंतपुर", "guntur": "गुन्तुर्", "chitoor": "चित्तूर", "rural": "ग्रामीण", "telugu": "तेलगु", "english": "अंग्रेज़ी", "kannada":"कन्नड़"}


marathiWord = {"anantapur": "अनंतपुर", "guntur": "गुन्तुर्", "chitoor": "चित्तूर", "rural": "ग्रामीण", "telugu": "तेलगु", "english": "अंग्रेज़ी", "kannada":"कन्नड़"}



# print(len(telugu_nums))
# print(len(telugu_letters))

# print(telugu_nums[16])
# print(sp_nums[6])
# print(telugu_letters['a'])

In [16]:
# dfile -> data from db
# pfile -> data from pdf

def getData(dfile,pfile):
    data = pickle.load(open(dfile,'rb'))
    df = pickle.load(open(pfile,'rb'))
    
    one =data['udise_sch_code'].to_list()
    two =df['Code'].to_list()
    two = [int(t) for t in two]
    
    intersection = sorted(set(one).intersection(set(two)))

    '''
    print("No. of codes in (1)DB - ", len(set(one)), " ,(2)PDF - ", len(set(two)))
    print("No. of codes NOT in PDF - ", len(set(one)-set(two)))
    print("No. of codes NOT in DB - ", len(set(two)-set(one)))
    print("No. of codes in BOTm - ", len(intersection))
    print("\n")
    '''
    return data, df, intersection

In [17]:
# సంబంధించినవి.
data = 'guntur'
print(transliterate(data, sanscript.ITRANS, sanscript.TELUGU))
print(transliterate(data, sanscript.ITRANS, sanscript.DEVANAGARI))

గున్తుర్
गुन्तुर्


In [18]:
# To TELUGU TRANSLITERATION
# Language specific code 

#case 0 (end) - pick ONLY from telugu_nums and NO మంది at the end
#case 1 (normal) - pick from telugu_nums and add మంది at the end or sp_nums 
def numToTelugu(Int, case=1):
    Int = int(Int)
    
    # Flag to denote if picking from sp_nums
    sp = 0
    
    if Int > 100:
        num =str(Int)
        
    # elif 0<Int<8 and case==1:
        # sp = 1
        # num =sp_nums[Int]
        
    elif Int<=20 or Int in [20,30,40,50,60,70,80,90,100]:
        num =marathi_nums[Int]
    
    else:
        ones =Int%10
        tens = int((Int/10))*10
        num =marathi_nums[tens]+' '+marathi_nums[ones]
        
    if case and not sp:
        # num+=' మంది'
        num += ''
        
    return num+' '


def charsToTelugu(abbr,f):
    tChars =''
    
    for c in abbr:
        if c.isalpha():
            c =marathi_letters[c]+f
        tChars+=c
            
    return tChars

def preprocess(phrase):
    new = ''
    for c in phrase:
        if not c.isalpha() and c not in ['.','_']:
            c =' '+c+' '
        new+=c
    
    return new


def toTelugu(phrase):
    tPhrase = ""
    
    phrase = phrase.lower()
    phrase = preprocess(phrase)
    
    words = phrase.split()
    
    spl = ['mpp','zpp','mpps','zphs','mpups','mjpapbcwrs','kgbv','aprei','nclp','icse','govt','cbse','dnt','apswrei','ssa','kgbvs']
    
    dot = '.'
    other = ''
    
    for word in words:
        tPhrase+=' '
        
        if word in marathiWord.keys():
            tPhrase+=marathiWord[word]
        
        elif word in spl or len(word)<3:
            tAbbr = charsToTelugu(word, other)
            tPhrase+=tAbbr
        
        elif '.' in word or '_' in word:
            part =""
            for c in word:
                if c=='.':
                    if len(part)==1:
                        tPhrase+= charsToTelugu(part,dot)
                    else:
                        tPhrase+= charsToTelugu(part,other)+'.'
                    part =""
                elif c=='_':
                    tPhrase+=charsToTelugu(part,other)+'_'
                    part =""
                elif c==' ':
                    if len(part)<3:
                        tPhrase+= charsToTelugu(part,other)
                    else:
                        tPhrase+=transliterate(part, sanscript.ITRANS, sanscript.DEVANAGARI)+' '
                    part = ""
                else:
                    part+=c
                    
            if part:
                if len(part)<3 or "_" in word:
                    tPhrase+= charsToTelugu(part,other)
                else:
                    tPhrase+=transliterate(part, sanscript.ITRANS, sanscript.DEVANAGARI)+' '
                    
        else:    
            tWord =transliterate(word, sanscript.ITRANS, sanscript.DEVANAGARI) 
            tPhrase+=tWord+' '
    
#     print(phrase, tPhrase.strip())
#     print(tPhrase.strip())
    
    return tPhrase.strip()+' '


In [19]:
# Language specific code

def getIntro(details):
    code = str(details[3]).strip()
    name = toTelugu(details[4].strip())
    village = toTelugu(details[2].strip())
    district = toTelugu(details[0].strip())
    block = toTelugu(details[1].strip())
    
    introOpts = ["ఈ పాఠశాల "+village+"గ్రామం లో ఉన్నది. ఈ గ్రామం "+district+"జిల్లా లో "+block+"పరిధి నందు ఉన్నది. ",
                      "ఈ పాఠశాల "+village+"గ్రామం లో ఉన్నది. "+district+"జిల్లా నందు "+block+"పరిధి లో ఈ గ్రామం ఉన్నది. ",
                      district+"జిల్లా లో "+block+"పరిధి లో "+village+"గ్రామం నందు ఈ పాఠశాల ఉన్నది. ",
                      "ఈ పాఠశాల "+district+"జిల్లా లో "+block+"పరిధి లో గల "+village+"గ్రామము నందు ఉన్నది. "
                    ]

    introOpts = ["यह विद्यालय " + village + " गाँव में स्थित है। यह गाँव " + district + " जिले में" + block + " श्रेणी में है।"]

    introOpts = ["ही शाळा " + village + " गावात आहे। हे गाव " + district + " जिले मध्ये" + block + " श्रेणीत आहे।"]




                    
    
    intro = random.choice(introOpts)
    intro = intro+" शाळा एकात्मिक जिल्हा माहिती शिक्षण प्रणाली (U-DISE) कोड "+code+" आहे।"
    
    return intro

# Returns Student's Type and Count of Students
def getStudentStats(counts):
    stuType = counts[1].lower()
    bcount =numToTelugu(counts[2])
    gcount = numToTelugu(counts[3])
    total = numToTelugu(int(counts[2])+int(counts[3]))
    
    sp=' '
    pick = 1
    
    
    if 'co' in stuType:
        opts1 =["ఇది ఒక కో-ఎడ్యుకేషనల్ పాఠశాల. ",
                    "ఇది ఒక బాల బాలికల పాఠశాల. " 
                   ]
        opts1 =["यह एक सह-शिक्षा विद्यालय है। ",
                    "यह एक लड़कों का और लड़कियों का स्कूल है। "]
        opts1 =["ही एक सह-शैक्षणिक शाळा आहे। ",
                    "ही मुले आणि मुलींची शाळा आहे। "]



        opts2 = ["ఇక్కడ "+bcount+"బాలురు "+gcount+"బాలికలు, మొత్తం "+total+"విద్యార్థులు చదువుతున్నారు. ",
                     "ఈ పాఠశాల లో మొత్తం "+total+"విద్యార్థులు చదువుతున్నారు. వీరిలో "+bcount+"బాలురు "+gcount+"బాలికలు ఉన్నారు. ",
                     "మొత్తం "+total+"విద్యార్థులలో "+bcount+"బాలురు "+gcount+"బాలికలు ఈ పాఠశాల లో చదువుతున్నారు. "
                    ]
        opts2 = ["इस स्कूल में "+bcount+" बालक और "+gcount+" बालिकाएं है और कुल "+total+" विद्यार्थी है"]
        opts2 = ["या शाळेत "+bcount+" मूल आणि "+gcount+" मुली आणि एकूण "+total+" विद्यार्थी आहे"]





        
    elif 'boy' in stuType:
        opts1 = ["ఇది ఒక బాలుర పాఠశాల. ",
                     "ఈ పాఠశాల లో బాలురకు మాత్రమే ప్రవేశం ఉన్నది. ",
                     "బాలురకు మాత్రమే ఈ పాఠశాల నందు ప్రవేశం ఉన్నది. "
                    ]
        opts1 = ["यह लड़कों का स्कूल है।"]
        opts1 = ["ती मुलांची शाळा आहे।"]

        opts2 = ["ఇక్కడ చదువుతున్న బాలుర సాంఖ్య "+numToTelugu(counts[2],0)+". ", # 0 signifies, pick only numbers normal numbers (end case)
                     "ఈ పాఠశాల లో "+bcount+"బాలురు చదువుతున్నారు. ",
                      bcount+"బాలురు ఈ పాఠశాల లో చదువుతున్నారు. "
                    ]
        opts2 = ["यहां पढ़ने वाले लड़कों की संख्या "+ bcount +" है। "]
        opts2 = ["येथे शिकणार्‍या मुलांची संख्या "+ bcount +" आहे। "]


        
    elif 'girl' in stuType:
        opts1 = ["ఇది ఒక బాలికల పాఠశాల. ",
                     "ఈ పాఠశాల లో బాలికలకు మాత్రమే ప్రవేశం ఉన్నది. ",
                     "బాలికలకు మాత్రమే ఇక్కడ ప్రవేశం ఉన్నది. "
                  ]
        opt1 = ["यह एक लड़कियों का स्कूल है। "]
        opt1 = ["ही मुलींची शाळा आहे। "]

        opts2 = ["ఇక్కడ చదువుతున్న బాలికల సాంఖ్య "+numToTelugu(counts[3],0)+". ",
                     "ఈ పాఠశాల లో "+gcount+"బాలికలు చదువుతున్నారు. ",
                      gcount+"బాలికలు ఇక్కడ చదువుతున్నారు. "
                    ]
        opts2 = [gcount +" लड़कियां इस स्कूल में पढ़ रही हैं।"]
        opts2 = [gcount +" या शाळेत शिकणार्‍या मुली।"]


    else:
        pick = 0
        
    if pick:
        line1 = random.choice(opts1)
        line2 = random.choice(opts2)
        studentStats =line1+line2
    else:
        studentStats = ''
        
    return studentStats

#Returns Teacher's Stats and Reference
def getTeacherStats(counts):
    sp = ' '
    mint = int(counts[5])
    fint =int(counts[6])
    mcount = numToTelugu(mint)
    fcount = numToTelugu(fint)
    
    # if only male OR female teachers.
    if mint>0 and fint==0:
        line ="ఈ పాఠశాల లో "+mcount+"ఉపాధ్యాయులు ఉన్నారు, వీరిలో అందరూ  ఉపాధ్యాయులే. "
        line = "स्कूल में "+ mcount +" शिक्षक हैं, जिनमें से सभी पुरुष हैं।"
        line = "शाळेमध्ये "+ mcount +" शिक्षक आहेत, जे सर्व पुरुष आहेत।"
    elif fint>0 and mint==0:
        line ="ఈ పాఠశాల లో "+fcount+"ఉపాధ్యాయులు ఉన్నారు, వీరిలో అందరూ ఉపాధ్యాయినులే. "
        line = "स्कूल में "+ mcount +" शिक्षिका  हैं, वह सभी महिलाएं हैं "
        line = "शाळेमध्ये"+ mcount +" शिक्षक आहेत, त्या सर्व स्त्रिया आहेत।"
        
    
    #Ekopadhya school
    elif mint==1 and fint==0:
        line ="ఇది ఒక ఏకోపాధ్యాయ పాఠశాల. ఈ పాఠశాల లో ఒక ఉపాధ్యాయుడు ఉన్నారు. "
        line ="यह वन-टीचर स्कूल है। इस स्कूल में एक शिक्षक है। "
        line ="ही एक शिक्षकांची शाळा आहे. या शाळेत एक शिक्षक आहे। "
    elif fint==1 and mint==0:
        line ="ఇది ఒక ఏకోపాధ్యాయ పాఠశాల. ఈ పాఠశాల లో ఒక ఉపాధ్యాయిని ఉన్నారు. "
        line ="यह वन-टीचर स्कूल है। इस स्कूल में एक महिला शिक्षक है। "
        line ="ही एक शिक्षकांची शाळा आहे. या शाळेत एक महिला शिक्षिका आहे। "
    
    else:
        total = numToTelugu(mint+fint)
        opts =["ఈ పాఠశాల లో "+mcount+"ఉపాధ్యాయులు "+fcount+"ఉపాధ్యాయినులే, మొత్తం "+total+"ఉపాధ్యాయులు ఉన్నారు. ",
                   "ఈ పాఠశాల లో మొత్తం "+total+"ఉపాధ్యాయులు ఉన్నారు. వీరిలో "+mcount+"ఉపాధ్యాయులు "+fcount+"ఉపాధ్యాయినులు ఉన్నారు. ",
                   "మొత్తం "+total+"ఉపాధ్యాయులు "+mcount+"ఉపాధ్యాయులు "+fcount+"ఉపాధ్యాయినులు ఈ పాఠశాల లో ఉన్నారు. "
                 ]
        opts = ["इस स्कूल में कुल " +total+ " अध्यापक है जिनमे से " +mcount+ " पुरुष और " +fcount+ " महिलाएं है "]
        opts = ["या शाळेत एकूण " +total+ " ज्याचे शिक्षक " +mcount+ " पुरुष आणि " +fcount+ " स्त्रिया आहेत"]


        line = random.choice(opts)
            
    return line
    
def getDesc(details):
    name =details[4].strip()
    mgnt = details[6].strip()
    
    line = ''
    if "Pvt.Aided" in mgnt:
        line = "यह एक निजी सहायता प्राप्त स्कूल है। "
        line = "ही खासगी अनुदानित शाळा आहे। "
    elif "Pvt.Unaided" in mgnt:
        line = "ఇది ఒక ప్రైవేటు పాఠశాల."
        line = "यह एक निजी स्कूल है। "
        line = "ती एक खासगी शाळा आहे। "
    elif "KGBV" in mgnt:
        line = "ఇది ఒక కస్తూర్బా గాంధీ బాలికా విద్యాలయ (KGBV). "
        line = "यह एक कस्तूरबा गांधी बालिका विद्यालय (केजीबीवी) है। "
        line = "हे कस्तुरबा गांधी बालिका विद्यालय आहे (केजीबीव्ही)। "
    
    elif "MPPS" in name:
        line = "ఇది ఒక మండల ప్రజా పరిషత్ ప్రాథమిక పాఠశాల (MPPS). "
        line = "यह एक मंडल प्रजा परिषद प्राथमिक विद्यालय (एमपीपीएस) है। "
        line = "ती मंडल प्रजा परिषद प्राथमिक शाळा (एमपीपीएस) आहे। "
    elif "MPPUS" in name:
        line = "ఇది ఒక మండల ప్రజా పరిషత్ ప్రాథమికోన్నత పాఠశాల (MPUPS). "
        line = "यह एक मंडल प्रजा परिषद प्राथमिक विद्यालय (MPUPS) है। "
        line = "ही मंडल प्रजा परिषद प्राथमिक शाळा आहे (एमपीयूपीएस)। "
    elif "ZPPS" in name:
        line = "ఇది ఒక జిల్లా పరిషత్ ప్రాథమిక పాఠశాల (ZPPS). "
        line = "यह एक जिला परिषद प्राथमिक विद्यालय (ZPPS) है। "
        line = "ती जिल्हा परिषद प्राथमिक शाळा (झेडपीपीएस) आहे। "
    elif "ZPHS" in name:
        line = "ఇది ఒక జిల్లా పరిషత్ ఉన్నత పాఠశాల (ZPHS). "
        line = "यह एक जिला परिषद हाई स्कूल (ZPHS) है। "
        line = "हे जिल्हा परिषद हायस्कूल आहे (झेडपीएचएस)। "
        
    return line

def getInfo(data, df, Id):
    details = data.loc[data['udise_sch_code']==Id].values.tolist()[0]
    counts = df.loc[df['Code']==str(Id)].values.tolist()[0]
    
    return details, counts

def getClass(lo, hi):
    if lo==1:
        line = "ఇక్కడ విద్యార్థులు "+class_nums[hi] +" తరగతి వరకూ విద్యని అభ్యసించవచ్చు. "
        line = "यहां पर छात्र " +class_nums[hi]+ " कक्षा तक शिक्षा प्राप्त कर सकते हैं | "
        line = "येथील विद्यार्थी " +class_nums[hi]+ " वर्ग पर्यंत शिक्षण प्राप्त करू शकता | "
    else:
        line = "ఇందులో "+class_nums[lo]+" నుండి "+class_nums[hi]+" తరగతి వరకు విద్యాబోధన కలదు. "
        line = "इसमें " +class_nums[lo]+ "से" +class_nums[hi]+ " कक्षा तक की शिक्षा शामिल है। "
        line = "या मध्ये " +class_nums[lo]+ "सपासून" +class_nums[hi]+ " वर्ग शिक्षण समाविष्ट आहे। "
    
    return line

def write(data, df, Id):
    article =""
    # Respective row details from pdf and excel data
    details, counts = getInfo(data, df, Id)
    
    #Sentence for name, village, district and block
    intro = getIntro(details)
    
    #Management
    management =toTelugu(details[6].strip())
    opts =[management+"నిర్వహణ లో ఈ పాఠశాల ఉన్నది. ",
               "ఈ పాఠశాల "+management+"నిర్వహణ లో పని చేస్తుంది. "
             ]
    opts = ["यह स्कूल "+ management +" के प्रबंधन के तहत संचालित होता है"]
    opts = ["ही शाळा "+ management +" च्या व्यवस्थापनाखाली चालते"]
    descLine =getDesc(details)
    oLine =random.choice(opts)+descLine
    
    #Classes
    desc = details[5].strip()
    p = re.compile(r'[A-Z]+(\d+)-(\d+)')
    m =re.match(p, desc)
    lo =int(m.group(1)); hi =int(m.group(2))
    cLine = getClass(lo, hi)
    
    #School Medium
    medium = toTelugu(details[7].strip())
    opts = [medium+"మాధ్యమం లో "+random.choice(["పాఠాలు ", "పాఠములు "])+"బోధిస్తారు. ",
                "ఈ పాఠశాల లో "+medium+"మాధ్యమం లో "+random.choice(["పాఠాలు ", "పాఠములు "])+"బోధిస్తారు. "
              ]
    opts = ["इस स्कूल में विद्यार्थियों को "+medium+" भाषा के माध्यम से पढ़ाया जाता ह। "]     
    opts = ["या शाळेतील विद्यार्थी "+medium+" माध्यमातून भाषा शिकविली जाते। "] 
    mLine = random.choice(opts)
    
    #School Type & student's counts
    studentStats = getStudentStats(counts)
    
    #Teachers counts + Reference Line
    teacherStats = getTeacherStats(counts)
    
    #Source
#     source ="ఈ వివరములు 2011-12 విద్యా సంవత్సరమునకు సంబంధించినవి. ఇవి నేషనల్ యూనివర్శిటీ ఆఫ్ ఎడ్యుకేషనల్ ప్లానింగ్ అండ్ అడ్మినిస్ట్రేషన్ (NUEPA) జరిపిన సర్వే ఆధారముగా ఇవ్వబడినవి. " 
    source = "ఈ వివరములు నేషనల్ యూనివర్శిటీ ఆఫ్ ఎడ్యుకేషనల్ ప్లానింగ్ అండ్ అడ్మినిస్ట్రేషన్ (NUEPA) జరిపిన సర్వే ఆధారముగా 2011-12 విద్యా సంవత్సరమునకు సంబంధించినవి."
    source = "ये विवरण 2011-12 के शैक्षणिक वर्ष के लिए नेशनल यूनिवर्सिटी ऑफ एजुकेशनल प्लानिंग एंड एडमिनिस्ट्रेशन (एनयूईपीए) द्वारा किए गए सर्वेक्षण पर आधारित हैं। "
    source = "नॅशनल युनिव्हर्सिटी ऑफ एज्युकेशनल प्लानिंग अँड अ‍ॅडमिनिस्ट्रेशन (एनयूईपीए) शैक्षणिक वर्ष २०११-१२ साठी केलेल्या सर्वेक्षणानुसार हे तपशील आहेत। "
    
    
    article =intro+oLine+cLine+mLine+studentStats+teacherStats+source
    print(article)
    
    return article


In [20]:
dfile_head = "DBpckls/"
pfile_head = "PDFpckls/"

dist = "ANANTAPUR"
data, df, Ids = getData(dfile_head+dist, pfile_head+dist)

In [21]:
#DEBUG TO GENERATE ONLY ONE ARTICLE
article=write(data, df, 28222501207)
print(article)
print(len(article.split()))

ही शाळा कक्कलपल्ले  ( ग्रामीण )  गावात आहे। हे गाव अनंतपुर  जिले मध्येअनंतपुर  श्रेणीत आहे। शाळा एकात्मिक जिल्हा माहिती शिक्षण प्रणाली (U-DISE) कोड 28222501207 आहे।ही शाळा एमपीपी_जेडपीपी स्चोओल्स्  च्या व्यवस्थापनाखाली चालतेयेथील विद्यार्थी आठवीं वर्ग पर्यंत शिक्षण प्राप्त करू शकता | या शाळेतील विद्यार्थी तेलगु  माध्यमातून भाषा शिकविली जाते। ही एक सह-शैक्षणिक शाळा आहे। या शाळेत 130  मूल आणि 170  मुली आणि एकूण 300  विद्यार्थी आहेया शाळेत एकूण ग्यारह  ज्याचे शिक्षक चार  पुरुष आणि सात  स्त्रिया आहेतनॅशनल युनिव्हर्सिटी ऑफ एज्युकेशनल प्लानिंग अँड अ‍ॅडमिनिस्ट्रेशन (एनयूईपीए) शैक्षणिक वर्ष २०११-१२ साठी केलेल्या सर्वेक्षणानुसार हे तपशील आहेत। 
ही शाळा कक्कलपल्ले  ( ग्रामीण )  गावात आहे। हे गाव अनंतपुर  जिले मध्येअनंतपुर  श्रेणीत आहे। शाळा एकात्मिक जिल्हा माहिती शिक्षण प्रणाली (U-DISE) कोड 28222501207 आहे।ही शाळा एमपीपी_जेडपीपी स्चोओल्स्  च्या व्यवस्थापनाखाली चालतेयेथील विद्यार्थी आठवीं वर्ग पर्यंत शिक्षण प्राप्त करू शकता | या शाळेतील विद्यार्थी तेलगु  माध्यमातून भाषा शिकविली जाते। ही एक सह-शैक

In [22]:
def customArticle(order, data, df, Id):
    article =""
    # Respective row details from pdf and excel data
    details, counts = getInfo(data, df, Id)
    
    #Sentence for name, village, district and block
    intro = getIntro(details)
    
    #Management
    management =toTelugu(details[6].strip())
    opts =[management+"నిర్వహణ లో ఈ పాఠశాల ఉన్నది. ",
               "ఈ పాఠశాల "+management+"నిర్వహణ లో పని చేస్తుంది. "
             ]
    opts = ["यह स्कूल "+ management +" के प्रबंधन के तहत संचालित होता है"]
    opts = ["ही शाळा "+ management +" च्या व्यवस्थापनाखाली चालते"]
    descLine =getDesc(details)
    oLine =random.choice(opts)+descLine
    
    #Classes
    desc = details[5].strip()
    p = re.compile(r'[A-Z]+(\d+)-(\d+)')
    m =re.match(p, desc)
    lo =int(m.group(1)); hi =int(m.group(2))
    cLine = getClass(lo, hi)
    
    #School Medium
    medium = toTelugu(details[7].strip())
    opts = [medium+"మాధ్యమం లో "+random.choice(["పాఠాలు ", "పాఠములు "])+"బోధిస్తారు. ",
                "ఈ పాఠశాల లో "+medium+"మాధ్యమం లో "+random.choice(["పాఠాలు ", "పాఠములు "])+"బోధిస్తారు. "
              ]
    opts = ["इस स्कूल में विद्यार्थियों को "+medium+" भाषा के माध्यम से पढ़ाया जाता ह। "]
    opts = ["या शाळेतील विद्यार्थी "+medium+" माध्यमातून भाषा शिकविली जाते। "] 
    mLine = random.choice(opts)
    
    #School Type & student's counts
    studentStats = getStudentStats(counts)
    
    #Teachers counts + Reference Line
    teacherStats = getTeacherStats(counts)
    
    #Source
    #source ="ఈ వివరములు 2011-12 విద్యా సంవత్సరమునకు సంబంధించినవి. ఇవి నేషనల్ యూనివర్శిటీ ఆఫ్ ఎడ్యుకేషనల్ ప్లానింగ్ అండ్ అడ్మినిస్ట్రేషన్ (NUEPA) జరిపిన సర్వే ఆధారముగా ఇవ్వబడినవి. " 
    source = "ఈ వివరములు నేషనల్ యూనివర్శిటీ ఆఫ్ ఎడ్యుకేషనల్ ప్లానింగ్ అండ్ అడ్మినిస్ట్రేషన్ (NUEPA) జరిపిన సర్వే ఆధారముగా 2011-12 విద్యా సంవత్సరమునకు సంబంధించినవి."
    source = "ये विवरण 2011-12 के शैक्षणिक वर्ष के लिए नेशनल यूनिवर्सिटी ऑफ एजुकेशनल प्लानिंग एंड एडमिनिस्ट्रेशन (एनयूईपीए) द्वारा किए गए सर्वेक्षण पर आधारित हैं। "
    
    source = "नॅशनल युनिव्हर्सिटी ऑफ एज्युकेशनल प्लानिंग अँड अ‍ॅडमिनिस्ट्रेशन (एनयूईपीए) शैक्षणिक वर्ष २०११-१२ साठी केलेल्या सर्वेक्षणानुसार हे तपशील आहेत। "
   
    # Write in custom order
    article = ""
    article_parts =[intro, oLine, cLine, mLine, studentStats, teacherStats, source]
    
    for n in order:
        article+=article_parts[int(n)-1]
    
    
    return article



In [23]:
details, counts = getInfo(data, df, 28222501207)


In [24]:
article=customArticle("1234567", data, df, 28222501207)
print(article)
print(len(article.split()))

ही शाळा कक्कलपल्ले  ( ग्रामीण )  गावात आहे। हे गाव अनंतपुर  जिले मध्येअनंतपुर  श्रेणीत आहे। शाळा एकात्मिक जिल्हा माहिती शिक्षण प्रणाली (U-DISE) कोड 28222501207 आहे।ही शाळा एमपीपी_जेडपीपी स्चोओल्स्  च्या व्यवस्थापनाखाली चालतेयेथील विद्यार्थी आठवीं वर्ग पर्यंत शिक्षण प्राप्त करू शकता | या शाळेतील विद्यार्थी तेलगु  माध्यमातून भाषा शिकविली जाते। ही एक सह-शैक्षणिक शाळा आहे। या शाळेत 130  मूल आणि 170  मुली आणि एकूण 300  विद्यार्थी आहेया शाळेत एकूण ग्यारह  ज्याचे शिक्षक चार  पुरुष आणि सात  स्त्रिया आहेतनॅशनल युनिव्हर्सिटी ऑफ एज्युकेशनल प्लानिंग अँड अ‍ॅडमिनिस्ट्रेशन (एनयूईपीए) शैक्षणिक वर्ष २०११-१२ साठी केलेल्या सर्वेक्षणानुसार हे तपशील आहेत। 
92


ఈ పాఠశాల కక్కలపల్లే  ( రూరల్  ) గ్రామం లో ఉన్నది. అనంతపూర్ జిల్లా నందు అనంతపూర్ పరిధి లో ఈ గ్రామం ఉన్నది. ఈ పాఠశాల యొక్క ఏకీకృత జిల్లా సమాచార విద్యా వ్యవస్థ (U-DISE) కోడ్ 28222501207. 
ఈ పాఠశాల ఎంపిపి_జెడ్పిపి స్చోఓల్స్ నిర్వహణ లో పని చేస్తుంది. 
ఇక్కడ విద్యార్థులు ఎనిమిదవ తరగతి వరకూ విద్యని అభ్యసించవచ్చు. 
ఈ పాఠశాల లో తేలుగు మాధ్యమం లో పాఠాలు బోధిస్తారు. 
ఇది ఒక కో-ఎడ్యుకేషనల్ పాఠశాల. ఇక్కడ 130 మంది బాలురు 170 మంది బాలికలు, మొత్తం 300 మంది విద్యార్థులు చదువుతున్నారు. 
ఈ పాఠశాల లో మొత్తం పదకొండు మంది ఉపాధ్యాయులు ఉన్నారు. వీరిలో నలుగురు ఉపాధ్యాయులు ఏడుగురు ఉపాధ్యాయినులు ఉన్నారు. 
ఈ వివరములు నేషనల్ యూనివర్శిటీ ఆఫ్ ఎడ్యుకేషనల్ ప్లానింగ్ అండ్ అడ్మినిస్ట్రేషన్ (NUEPA) జరిపిన సర్వే ఆధారముగా 2011-12 విద్యా సంవత్సరమునకు సంబంధించినవి.

In [25]:
#GENERATING MULTIPLE ARTICLES
for dist in ["ANANTAPUR", "CHITTOOR","GUNTUR"]:
    print(dist)
    data, df, Ids = getData(dfile_head+dist, pfile_head+dist)

    failed = []
    for Id in Ids:
        try:
            article =write(data, df, Id)
    #         print("pass")
        except:
            failed.append(Id)
            print('\t',Id)
        
    print('\t',failed)
    print('\t',len(failed))
    print('\t',len(Ids))

टी.ऐदेद्  च्या व्यवस्थापनाखाली चालतेही खासगी अनुदानित शाळा आहे। येथील विद्यार्थी पाचवा वर्ग पर्यंत शिक्षण प्राप्त करू शकता | या शाळेतील विद्यार्थी तेलगु  माध्यमातून भाषा शिकविली जाते। ही मुले आणि मुलींची शाळा आहे। या शाळेत पाच  मूल आणि दहा  मुली आणि एकूण  पंद्रह   विद्यार्थी आहेशाळेमध्येशून्य  शिक्षक आहेत, त्या सर्व स्त्रिया आहेत।नॅशनल युनिव्हर्सिटी ऑफ एज्युकेशनल प्लानिंग अँड अ‍ॅडमिनिस्ट्रेशन (एनयूईपीए) शैक्षणिक वर्ष २०११-१२ साठी केलेल्या सर्वेक्षणानुसार हे तपशील आहेत। 
ही शाळा बपत्ल  ( एम )  गावात आहे। हे गाव गुन्तुर्  जिले मध्येबपत्ल  श्रेणीत आहे। शाळा एकात्मिक जिल्हा माहिती शिक्षण प्रणाली (U-DISE) कोड 28175790127 आहे।ही शाळा पीवीटी.ऐदेद्  च्या व्यवस्थापनाखाली चालतेही खासगी अनुदानित शाळा आहे। येथील विद्यार्थी पाचवा वर्ग पर्यंत शिक्षण प्राप्त करू शकता | या शाळेतील विद्यार्थी तेलगु  माध्यमातून भाषा शिकविली जाते। ही एक सह-शैक्षणिक शाळा आहे। या शाळेत 105  मूल आणि 120  मुली आणि एकूण 225  विद्यार्थी आहेया शाळेत एकूण आठ  ज्याचे शिक्षक चार  पुरुष आणि चार  स्त्रिया आहेतनॅशनल युनिव्हर्सिटी ऑफ 

In [26]:
def main():
    dfile_head = "DBpckls/"
    pfile_head = "PDFpckls/"
    
    high=0
    big =''
    for dist in ["ANANTAPUR", "CHITTOOR", "GUNTUR"]:
        data, df, ids = getData(dfile_head+dist, pfile_head+dist)
        for Id in ids:
            atricle =write(data, df, Id)

